packages/core/strapi/lib/types/utils/expression.d.ts

```diff
@@ -8,6 +8,10 @@ export type IsNever<TValue> = [TValue] extends [never] ? True : False;
 
 export type IsNotNever<TValue> = Not<IsNever<TValue>>;
 
+export type IsTrue<TValue> = [TValue] extends [True] ? True : False;
```

Comment:
Do you have some kind of doc that explain why we need to do this ?

Reply:
I searched a bit online but didn't find anything. I'll try to dig in the TS type check codebase later and see if I can draft an explanation.

I've come to this mainly by xp & experimenting the tuple/types behaviors

Reply:
Yeah let's write an article on that if we find the explanations someday :D